From 7d227d8dd1c30398d482e9ab1a76ae478d187132 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Mon, 12 Feb 2018 10:09:57 +0200
Subject: [PATCH 3389/5242] MLK-17634-3: drm: edid: fix hdr infoframe creation
 routine

commit  0867dde01c3a524679b3cb7387d40e1abcbe55c1 from
https://source.codeaurora.org/external/imx/linux-imx.git

The frame->type was overwritten, instead of setting the frame->metadata_type
field.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/drm_edid.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/drm_edid.c b/drivers/gpu/drm/drm_edid.c
index 4d67b0e..0b56e7d 100644
--- a/drivers/gpu/drm/drm_edid.c
+++ b/drivers/gpu/drm/drm_edid.c
@@ -4957,9 +4957,8 @@ void drm_set_preferred_mode(struct drm_connector *connector,
 
 	frame->length = sizeof(struct hdr_static_metadata);
 
-
 	frame->eotf = hdr_source_metadata->eotf;
-	frame->type = hdr_source_metadata->type;
+	frame->metadata_type = hdr_source_metadata->type;
 
 	for (i = 0; i < 3; i++) {
 		frame->display_primaries_x[i] =
-- 
1.7.9.5

